index.js

```diff
@@ -1,253 +1,37 @@
-    propStream = require('prop-stream'),
-    settings = require('pelias-config').generate(),
-    dbclient = require('pelias-dbclient')(),
-    OsmiumStream = require('osmium-stream');
-
-var Backend = require('geopipes-elasticsearch-backend');
-
-// esclient.livestats();
-
-var backend = function( index, type ){
-  return new Backend( dbclient.client, index, type );
-};
-
-// @todo: extract this or refactor suggester to make it a stream factory
-var bun = require('bun');
-function generateSuggester(){
-  return bun([
-    suggester.streams.suggestable(),
-    suggester.streams.suggester( suggester.generators )
-  ]);
-}
-
-// use datapath setting from your config file
-var basepath = settings.imports.openstreetmap.datapath;
-var filename = settings.imports.openstreetmap.import[0].filename;
-var leveldbpath = settings.imports.openstreetmap.leveldbpath;
-
-// testing
-// basepath = '/media/hdd/osm/mapzen-metro';
-// filename = 'wellington.osm.pbf';
-
-var pbfFilePath = basepath + '/' + filename;
-// check pbf file exists
-try {
-  fs.statSync( pbfFilePath );
-} catch( e ){
-  console.error( 'failed to load pbf file:', pbfFilePath );
-  process.exit(1);
-}
-
-// check leveldb dir exists
-try {
-  fs.statSync( leveldbpath );
-} catch( e ){
-  console.error( 'failed to open:', leveldbpath );
-  process.exit(1);
-}
-
-// generic streams
-var stringify =                require('./stream/stringify');
-var devnull =                  require('./stream/devnull');
-var required =                 require('./stream/required');
-// var sporadic =                 require('./stream/sporadic');
+/**
+  The openstreetmap importer provides a way of parsing, mapping and
+  augmenting OSM data in to elasticsearch.
+**/
 
-// sinks
-var backend = {
-  es: {
-    osmnode:                  backend('pelias', 'osmnode'),
-    osmway:                   backend('pelias', 'osmway'),
-    geonames:                 backend('pelias', 'geoname'),
-    admin0:                   backend('pelias', 'admin0'),
-    admin1:                   backend('pelias', 'admin1'),
-    admin2:                   backend('pelias', 'admin2'),
-    local_admin:              backend('pelias', 'local_admin'),
-    locality:                 backend('pelias', 'locality'),
-    neighborhood:             backend('pelias', 'neighborhood')
-  },
-  level: {
-    osmnodecentroids:         multilevel( levelup( leveldbpath ), 'm' )
-  }
-};
-
-// forkers
-var osm_types =                require('./stream/osm_types');
-
-// osm
-var osm = {
-  way: {
-    denormalizer:              require('./stream/osm/way/denormalizer'),
-  },
-  any: {
-    hierarchyLookup:            require('./stream/osm/any/hierarchyLookup')
-  }
+var elasticsearch = require('pelias-dbclient'),
+    adminLookup = require('pelias-admin-lookup'),
+    suggester = require('pelias-suggester-pipeline'),
+    dbmapper = require('./stream/dbmapper');
+
+var osm = { pbf: {}, doc: {}, address: {}, tag: {} };
```

Comment:
in my opinion this would work better as a class, especially since it's being exported.